roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -164,50 +254,223 @@ def get_num_channels(self):
         return self._num_channels
 
     @staticmethod
-    def write_imaging(imaging, savepath):
-        pass
+    def add_devices(imaging, nwbfile, metadata):
+        # Devices
+        if 'Ophys' not in metadata:
+            metadata['Ophys'] = dict()
+        if 'Device' not in metadata['Ophys']:
+            metadata['Ophys']['Device'] = [{'name': 'Device'}]
+        # Tests if devices exist in nwbfile, if not create them from metadata
+        for dev in metadata['Ophys']['Device']:
+            if dev['name'] not in nwbfile.devices:
+                nwbfile.create_device(name=dev['name'])
+
+        return nwbfile
+
+    @staticmethod
+    def add_two_photon_series(imaging, nwbfile, metadata):
+        """
+        Auxiliary static method for nwbextractor.
+        Adds two photon series from imaging object as TwoPhotonSeries to nwbfile object.
+        """
+        if 'Ophys' not in metadata:
+            metadata['Ophys'] = {}
+
+        if 'Ophys' not in metadata or 'TwoPthotonSeries' not in metadata['Ophys']:
+            metadata['Ophys']['TwoPhotonSeries'] = [{'name': 'TwoPhotonSeries',
+                                                     'description': 'optical_series_description'}]
+        # Tests if ElectricalSeries already exists in acquisition
+        nwb_es_names = [ac for ac in nwbfile.acquisition]
+        opts = metadata['Ophys']['TwoPhotonSeries'][0]
+        if opts['name'] not in nwb_es_names:
+            # retrieve device
+            device = nwbfile.devices[list(nwbfile.devices.keys())[0]]
+
+            # create optical channel
+            if 'OpticalChannel' not in metadata['Ophys']:
+                metadata['Ophys']['OpticalChannel'] = [{'name': 'OpticalChannel',
+                                                        'description': 'no description',
+                                                        'emission_lambda': 500.}]
+
+            optical_channel = OpticalChannel(**metadata['Ophys']['OpticalChannel'][0])
```

Comment:
```suggestion
            optical_channel_args = dict(name='OpticalChannel', description='no description', emission_lambda=np.nan)
            
            if 'OpticalChannel' in metadata['Ophys']:
                optical_channel_args.update(metadata['Ophys']['OpticalChannel'][0])

            optical_channel = OpticalChannel(**optical_channel_args)
```

If we make up placeholder data, it needs to be clear that we are doing that. In this case that means using `np.nan`. Using `500.` will inevitably lead to us converting data incorrectly!

Reply:
also this approach allows for the args to OpticalChannel to be partially filled in